eval.go

```diff
@@ -370,6 +370,11 @@ nextFileMatch:
 
 		// Prefer earlier docs.
 		fileMatch.addScore("doc-order", scoreFileOrderFactor*(1.0-float64(nextDoc)/float64(len(d.boundaries))), opts.DebugScore)
+
+		if len(d.ranks) > int(nextDoc) {
+			fileMatch.ranks = d.ranks[nextDoc]
```

Comment:
this seems very sub-optimal? We are no longer taking into account query based scoring? We really should think of a way to combine a document ranks with how a query matches a result. I worry about our result quality going down the toilet to be honest.

Reply:
That's a bit extreme, isn't it :-) We don't ignore the score, it is just limited to reordering matches within a file. I don't have a solution for mixing a rank-vector and a match score yet, but this is something we can iterate on. I am still building this with a local demo in mind.

Reply:
ok if it is for the demo that is fine. but in practice I imagine only relying on document rank will lead to worse ranking. Things like scoring based on query alignment and symbol matches seems super important.

Reply:
updated the PR to use RRF to blend the rankings of the 2 scoring functions (offline ranking, match scores)